### PR TITLE
chore: bring example up to date with code

### PR DESCRIPTION
**Pull request by DanielMSchmidt**
Hey there, I made a quick PR to bring the threaded example in sync with the actual code. I also added a github action so it stays in sync, I hope that's okay :) 

When I run this I get

```
     Running `/Users/danielschmidt/fun/nokhwa/target/debug/threaded-capture`
CameraInfo { human_name: "FaceTime HD Camera", description: "Apple Inc.: FaceTime HD Camera - AVCaptureDeviceTypeBuiltInWideAngleCamera, Unspecified f0", misc: "EAB7A68F-EC2B-4487-AADF-D8A91C1CB782", index: Index(0) }
thread 'main' panicked at 'called `Result::unwrap()` on an `Err` value: OpenDeviceError("0", "Not Found")', examples/threaded-capture/src/main.rs:39:6
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
```

Is this expected in the current RC state?

### CODE REVIEW

**Comment by DanielMSchmidt** [comment]
I think my panic stems from the usage of [AVCaptureDevice.devices()](https://developer.apple.com/documentation/avfoundation/avcapturedevice/1386237-devices) at [this part](https://github.com/DanielMSchmidt/nokhwa/blob/3f48d0fa7de6299afb399465b9f8068298f44572/nokhwa-bindings-macos/src/lib.rs#L915) that is used when creating a new camera. It is different to the implementation of [query_avfoundation](https://github.com/DanielMSchmidt/nokhwa/blob/3f48d0fa7de6299afb399465b9f8068298f44572/nokhwa-bindings-macos/src/lib.rs#L926) shouldn't it be the same?

**Comment by l1npengtul** [comment]
It should, it is probably an oversight as I have lost access to my mac, I will try to get to fixing this today if possible.

File: examples/threaded-capture/Cargo.toml [hunk 1]
```diff
@@ -14,4 +14,4 @@ default-features = false
 [dependencies.nokhwa]
 path = "../../../nokhwa"
 # EDIT THIS!
-features = ["input-v4l", "output-threaded"]
+features = ["input-avfoundation", "output-threaded"]
```
Comment (l1npengtul) [inline examples/threaded-capture/Cargo.toml]: This should be `input-native` in 0.10

File: examples/threaded-capture/src/main.rs [hunk 2]
```diff
@@ -14,27 +14,44 @@
...
+        println!("User said {}", granted);
+    });
     let cameras = query(ApiBackend::Auto).unwrap();
     cameras.iter().for_each(|cam| println!("{:?}", cam));
 
-    let mut threaded = ThreadedCamera::new(0, None).unwrap();
-    threaded.open_stream(callback).unwrap();
+    let format = RequestedFormat::with_formats(
+        RequestedFormatType::AbsoluteHighestFrameRate,
+        &[FrameFormat::GRAY],
```
Comment (l1npengtul) [inline examples/threaded-capture/src/main.rs]: This should be `new` with `::<RgbFormat>`

**Comment by DanielMSchmidt** [comment]
Fixed your comments, got the same error though :)

**Comment by l1npengtul** [comment]
Did you update the fork? I pushed a commit that should make the search implementations the same

**Comment by DanielMSchmidt** [comment]
No, I didn't. Now we are one step further:

```
     Running `/Users/danielschmidt/fun/nokhwa/target/debug/threaded-capture`
User said true
CameraInfo { human_name: "FaceTime HD Camera", description: "Apple Inc.: FaceTime HD Camera - AVCaptureDeviceTypeBuiltInWideAngleCamera, Unspecified f0", misc: "EAB7A68F-EC2B-4487-AADF-D8A91C1CB782", index: Index(0) }
thread 'main' panicked at 'called `Result::unwrap()` on an `Err` value: StructureError { structure: "FourCharCode", error: "Unknown FourCharCode 875704438" }', examples/threaded-capture/src/main.rs:41:6
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
```

**Comment by l1npengtul** [comment]
This is very weird, I added 420v support but it doesn't match for some reason...

I pushed a hacky fix, see if it works.

Keep in mind that the NV12 decoder is still broken so dont mind the distorted imagery :p

**Comment by DanielMSchmidt** [comment]
I ran it again (removing one leftover f) and this time I got

```
thread 'main' panicked at 'called `Result::unwrap()` on an `Err` value: SetPropertyError { property: "CameraFormat", value: "720x1280@30FPS, NV12 Format", error: "Not Found/Rejected/Unsupported" }', examples/threaded-capture/src/main.rs:41:6
```

Is this the error you expected from the nv12 decoder? (Do I understand it correctly that I won't get a proper image out until this decoder is fixed?)

**Comment by l1npengtul** [comment]
No, that is a different error.

Apologies for this trial and error, whatever I do my mac VM doesnt seem to want to execute my programs...

**Comment by DanielMSchmidt** [comment]
No problem at all, I'm always happy to help, this library will be an important piece for my side-project :) (There it will run on a raspberry pi though :D)

**Comment by l1npengtul** [comment]
Raspis are linux, so it should be well supported there at least :p

**Comment by l1npengtul** [comment]
Updated, try again please, thank you.

**Comment by DanielMSchmidt** [comment]
This time I get this even with RUST_BACKTRACE set


```
User said true
CameraInfo { human_name: "FaceTime HD Camera", description: "Apple Inc.: FaceTime HD Camera - AVCaptureDeviceTypeBuiltInWideAngleCamera, Unspecified f0", misc: "EAB7A68F-EC2B-4487-AADF-D8A91C1CB782", index: Index(0) }
fish: Job 1, 'env RUST_BACKTRACE=1 cargo run' terminated by signal SIGTRAP (Trace or breakpoint trap)
```

**Comment by DanielMSchmidt** [comment]
I rebased again and this time it's 

```
User said true
CameraInfo { human_name: "FaceTime HD Camera", description: "Apple Inc.: FaceTime HD Camera - AVCaptureDeviceTypeBuiltInWideAngleCamera, Unspecified f0", misc: "EAB7A68F-EC2B-4487-AADF-D8A91C1CB782", index: Index(0) }
fish: Job 1, 'env RUST_BACKTRACE=1 cargo run' terminated by signal SIGBUS (Misaligned address error)
```

**Comment by l1npengtul** [comment]
sorry, i wont be able to work on this PR for a bit... macos VM bit the dust and because its SAT week i dont have time to fix it...

**Comment by DanielMSchmidt** [comment]
No problem :)

**Comment by l1npengtul** [comment]
The latest 2 commits have fixed AVFoundation. The only thing standing between working 0.10 and right now is a broken NV12 decoding implementation, which I am working on :>

**Review by l1npengtul — APPROVED** [review]

**Comment by l1npengtul** [comment]
Thank you for contributing!

**Pull request merged**